store/src/java/com/zimbra/cs/zimlet/ProxyServlet.java

```diff
@@ -83,8 +88,11 @@ public class ProxyServlet extends ZimbraServlet {
     private static final String AUTH_PARAM = "auth";
     private static final String AUTH_BASIC = "basic";
 
+    private static boolean isRedirectStatus(int status) {
+        return status / 100 == 3; // 3xx Redirect
```

Comment:
in case of status 301
301/100 = 3.01
so 3.01 is not equal to 3, so in that case are we going to return false?

Reply:
It'd be a problem if either was a float, but because they're both integers, `301/100 == 3`. Added some tests to prove the point.

Reply:
aah good point, completely forgot I am reviewing java which is strict typed language unlike javascript :)